src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdLifetime.cs

```diff
@@ -34,6 +34,7 @@ public SystemdLifetime(IHostEnvironment environment, IHostApplicationLifetime ap
             ThrowHelper.ThrowIfNull(environment);
             ThrowHelper.ThrowIfNull(applicationLifetime);
             ThrowHelper.ThrowIfNull(systemdNotifier);
+            ThrowHelper.ThrowIfNull(loggerFactory);
```

Comment:
Note that these lifetime classes are only expected to be used inside the "Hosting" infrastructure, which guarantees a LoggingFactory is registered in DI. There is no need to make the parameters nullable, and fallback to a no-op logger, since a logger will always be registered in the intended use cases.

If callers are trying to use these classes outside of "Hosting", they can pass `NullLoggerFactory.Instance` themselves.